proto/regen/ecocredit/v1/tx.proto

```diff
@@ -273,6 +311,128 @@ message MsgCreateProjectResponse {
+  string project_id = 2;
+
+  // class_id is the identifier of the credit class which the project is
+  // applying to.
+  string class_id = 3;
+
+  // issuer is the address of the account that is an issuer of the credit class
+  // to whom the project is applying to for approval.
+  string issuer = 4;
+
+  // metadata is any arbitrary string with a maximum length of 256 characters
+  // that includes or references any metadata relevant to the application.
+  // This could be used as a digital reference to the actual contents of the application.
+  string metadata = 5;
+}
+
+// MsgSubmitClassApplicationResponse is the Msg/SubmitClassApplication response type.
+message MsgSubmitClassApplicationResponse {
+  // application_id is the identifier assigned to the application.
+  uint64 application_id = 1;
+}
+
+// MsgUpdateClassApplication is the Msg/UpdateClassApplication request type.
+message MsgUpdateClassApplication {
+  option (cosmos.msg.v1.signer) = "project_admin";
+
+  // project_admin is the address of the account that is the admin of the
+  // project which is updating its application to the credit class.
+  string project_admin = 1;
+
+  // application_id is the identifier of the application to update.
+  uint64 application_id = 2;
+
+  // metadata is any arbitrary string with a maximum length of 256 characters
+  // that includes or references metadata relevant to the application. If it
+  // is left empty, the existing metadata will be deleted.
+  string metadata = 3;
+}
+
+// MsgUpdateClassApplicationResponse is the Msg/UpdateClassApplication response type.
+message MsgUpdateClassApplicationResponse {}
+
+// MsgWithdrawClassApplication is the Msg/WithdrawClassApplication request type.
+message MsgWithdrawClassApplication {
+  option (cosmos.msg.v1.signer) = "project_admin";
+
+  // project_admin is the address of the account that is the admin of the
+  // project which is withdrawing its application to the credit class.
+  string project_admin = 1;
+
+  // application_id is the identifier of the application to withdraw.
+  uint64 application_id = 2;
+}
+
+// MsgWithdrawClassApplicationResponse is the Msg/WithdrawClassApplication response type.
+message MsgWithdrawClassApplicationResponse {}
+
+// MsgEvaluateClassApplication is the Msg/EvaluateClassApplication request type.
+message MsgEvaluateClassApplication {
+  option (cosmos.msg.v1.signer) = "issuer";
+
+  // issuer is the address of the account that is the issuer of the credit class
+  // which is evaluating the application.
+  string issuer = 1;
+
+  // application_id is the identifier of the application to evaluate.
+  uint64 application_id = 2;
+
+  // evaluation is the evaluation of the application.
+  ApplicationStatus evaluation = 3;
+
+  // reason is any arbitrary string with a maximum length of 256 characters
+  // that includes or references the reason for the approving, requesting changes
+  // to, or rejecting the application.
+  string reason = 4;
```

Comment:
The assumption here is that if it's longer than 256 characters, they could always provide a link to a longer response?

Reply:
Yes